File: app/src/main/java/com/mynotebook/englishonthego/MainActivity.java
package com.mynotebook.englishonthego;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;
import androidx.viewpager2.widget.ViewPager2;

import android.os.Bundle;
import android.view.View;

import com.mynotebook.englishonthego.databinding.ActivityMainBinding;
import com.mynotebook.englishonthego.ui.FragmentCategoryAdapter;
import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;

public class MainActivity extends AppCompatActivity  {

    private ActivityMainBinding binding;

    /**
     * Fragment objects declaration
     */
    private ViewPager2 viewPager;
    private FragmentStateAdapter fragmentStateAdapter;
    String[] tabsName = {"Lyric Search", "My Dictionary", "My Notes"};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        /**
         * Fragment related instantiations
         */
        viewPager = findViewById(R.id.pager);
        fragmentStateAdapter = new FragmentCategoryAdapter(this);
        viewPager.setAdapter(fragmentStateAdapter);
        TabLayout tabLayout = findViewById(R.id.tab_layout);
        new TabLayoutMediator(tabLayout, viewPager, (tab, position) ->
                tab.setText(tabsName[position]))
                .attach();
    }
}

File: app/src/main/java/com/mynotebook/englishonthego/MainActivity.kt
package com.mynotebook.englishonthego

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator
import com.google.android.material.tabs.TabLayoutMediator.TabConfigurationStrategy
import com.mynotebook.englishonthego.ui.FragmentCategoryAdapter
import com.mynotebook.englishonthego.utilities.SharedPrefUtils
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private var tabsName = arrayOf("Lyric Search", "Lyrics", "Dictionary", "Notes")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initTabs()
    }

    override fun onPause() {
        SharedPrefUtils.setLastTabPosition(this, pager.currentItem)
        super.onPause()
    }

    private fun initTabs() {
        val fragmentStateAdapter = FragmentCategoryAdapter(this)
        pager.adapter = fragmentStateAdapter
        pager.currentItem = SharedPrefUtils.getLastTabPosition(this)

        val tabLayout = findViewById<TabLayout>(R.id.tab_layout)
        pager.let {
            TabLayoutMediator(
                    tabLayout,
                    it,
                    TabConfigurationStrategy { tab: TabLayout.Tab, position: Int -> tab.text = tabsName[position] }
            ).attach()
        }
    }
}
